src/client/conn/http2.rs

```diff
@@ -90,20 +90,9 @@ impl<B> SendRequest<B> {
         futures_util::future::poll_fn(|cx| self.poll_ready(cx)).await
     }
 
-    /*
-    pub(super) async fn when_ready(self) -> crate::Result<Self> {
-        let mut me = Some(self);
-        future::poll_fn(move |cx| {
-            ready!(me.as_mut().unwrap().poll_ready(cx))?;
-            Poll::Ready(Ok(me.take().unwrap()))
-        })
-        .await
-    }
-
     pub(super) fn is_ready(&self) -> bool {
```

Comment:
A difference is that in 1.0, these are fully public. Well, this one, and the next one.

Reply:
I think I actually intended to make this `pub` and this was an oversight. Is this (and closed) intentionally not `pub` here but `pub` in 1.0? I can change is_closed or close this PR accordingly

Reply:
It just wasn't made public because there hasn't been a need. In that sense, I tend to wait till someone has a request, so as not to expose the API prematurely. 🤷

Reply:
Fair enough! Our use case is to use this as part of a connection pool; we are working to move to 1.0 to do this but were hoping to implement pooling first while we work on the rest of the migration; exposing this would enable that.